tests/suites/test_suite_pk.function

```diff
@@ -737,27 +737,35 @@ exit:
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_PK_PARSE_C:MBEDTLS_FS_IO */
+/* BEGIN_CASE depends_on:MBEDTLS_PK_PARSE_C:MBEDTLS_FS_IO:MBEDTLS_PSA_CRYPTO_C */
```

Comment:
I don't think that's correct. The function itself does not depend on `PSA_CRYPTO_C`. Only the parts that used to depend on `USE_PSA_CRYPTO` (and that you changed to `PSA_CRYPTO_CLIENT`) now depend on `PSA_CRYPTO_C`.

Also, we should change `USE_PSA_INIT/DONE` to `PSA_INIT/DONE()` (which do nothing if `PSA_CRYPTO_C` is not defined). This might apply to other test functions too.

Reply:
Yes, you're right. I also checked other functions I modified in this PR, but in all these cases I changed the overall function guard from `USE_PSA` to `CRYPTO_C` because those functions are heavily depending on opaque keys (not just a minor testing that can easily be excluded).
[This commit](https://github.com/Mbed-TLS/mbedtls/pull/8842/commits/fd4dcbfe8e428a5530121a215e4f822679aad5a2) should fix this problem :wink: